samples/boards/infineon/pulse_counter/src/main.c

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2023 Cypress Semiconductor Corporation (an Infineon company) or
+ * an affiliate of Cypress Semiconductor Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/counter.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/sys/printk.h>
+
+struct counter_alarm_cfg alarm_cfg;
+
+#define COUNTER_1  DEVICE_DT_GET(DT_NODELABEL(counter1_0))
+#define PULSE_PIN  1
+#define PULSE_NODE DEVICE_DT_GET(DT_NODELABEL(gpio_prt9))
+
+struct counter_top_cfg counter_top;
+int counter_top_userdata;
+
+static void cb_counter_top(const struct device *dev, void *user_data)
+{
+	int err;
+
+	int *p_data = (int *)user_data;
+
+	printk("Counter top reached. Resetting Count\n");
+	printk("Top Count = %d\n", *p_data);
+	*p_data = *p_data + 1;
+
+	counter_top.ticks = counter_top.ticks + 20;
+
+	err = counter_set_top_value(dev, &counter_top);
+	if (err != 0) {
+		printk("Counter settings failed with error code %d\n", err);
+	}
+}
+
+static void cb_alarm(const struct device *counter_dev, uint8_t chan_id, uint32_t ticks,
+		     void *user_data)
+{
+	int err;
+
+	printk("!! Alarm !!\n");
+	printk("!! Count = %u\n", ticks);
+
+	struct counter_alarm_cfg *p_alarm = (struct counter_alarm_cfg *)user_data;
```

Comment:
Same as in main, keep these at the top, not sure it's explicitly in the coding style but that's how most of the code is written.

Reply:
Fixed. Also changed in a couple other functions